docs/rfc/rfc-100-abci-vote-extension-propag.md

```diff
@@ -555,6 +577,144 @@ Then, we need to handle two new situations, roughly equivalent to cases (h.1) an
+to transition the chain from maintaining no history of vote extensions to requiring vote extensions.
+This parameter is an `int64` representing the first height where vote extensions
+will be required for votes to be considered valid.
+
+The initial value of this `ConsensusParam` is 0,
+which is also its implicit value in versions prior to ABCI 2.0,
+denoting that an extension-enabling height has not been decided yet.
+Once the upgrade to ABCI 2.0 has taken place,
+the value MAY be set to some height, *h<sub>e</sub>*,
+which MUST be higher than the current height of the chain.
+From the moment when the `ConsensusParam` > 0,
+for all heights *h ≥ h<sub>e</sub>*, the consensus algorithm will
+reject any votes that do not have vote extension data as invalid.
+Likewise, for all heights *h < h<sub>e</sub>*, any votes that *do* have vote extensions
+will be considered an error condition.
+Height *h<sub>e</sub>* is somewhat special, as calls to `PrepareProposal` MUST NOT
+have vote extension data, but all precommit votes in that height MUST carry a vote extension.
+Height *h<sub>e</sub> + 1* is the first height for which `PrepareProposal` MUST have vote
+extension data and all precommit votes in that height MUST have a vote extension.
+
+#### Upgrading and Transitioning to Vote Extensions
+
+Just after upgrading (via coordinated upgrade) to ABCI 2.0, vote extensions stay disabled,
+as the Application needs to decide on a future height to be set for transitioning to vote extensions.
+The earliest this can happen is *h<sub>u</sub> + 1*, where *h<sub>u</sub>* denotes the upgrade height,
+i.e., the height at which all nodes will start when they restart with the upgraded binary.
+
+Once a node reaches the configured height *h<sub>e</sub>*, the parameter is disallowed from changing.
+Vote extensions cannot flip from being required to being optional.
+This is enforced by the `ConsensusParam` validation logic. Forcing vote extensions to
+be required beyond the configured height simplifies the logic for transitioning
+from optional to required since all checks will only need to understand if the
+chain *ever* enabled vote extensions in the past. Additionally, the major known
+uses cases of vote extensions such as threshold decryption and oracle data will
+be *central* components of the applications that use vote extensions. Flipping
+vote extensions to be no longer required will fundamentally change the behavior
+of the application and is therefore not valuable to these applications.
+
+Additional discussion and implementation of this upgrade strategy can be found
+in GitHub [issue 8453][toggle-vote-extensions].
+
+We now explain the changes needed to key solutions/implementation proposed in previous sections.
+For simplicity, in any conditions comparing a height to *h<sub>e</sub>*, 
+if *h<sub>e</sub>* is 0 (not set yet) then the condition assumes *h<sub>e</sub> = ∞*.
```

Comment:
Essentially the considered height will always be considered as less than h_e and thus allow the code to proceed as if it does not need vote extensions?

Reply:
Exactly. It's a ruse to not complicate all the conditions that appear below

Reply:
If you think that's not clear enough, maybe we can add one sentence